src/coprocessor/statistics/analyze.rs

```diff
@@ -219,36 +261,57 @@ impl<S: Snapshot> SampleBuilder<S> {
             );
             self.col_len
         ];
-        while let Some(row) = self.data.next()? {
-            let row = row.take_origin()?;
-            let cols = row.get_binary_cols(&mut EvalContext::default())?;
-            let retrieve_len = cols.len();
-            let mut cols_iter = cols.into_iter();
-            if self.col_len != retrieve_len {
-                if let Some(v) = cols_iter.next() {
-                    pk_builder.append(&v);
+        let mut is_drained = false;
+        while !is_drained {
+            let result = self.data.next_batch(BATCH_MAX_SIZE);
+            is_drained = result.is_drained?;
+
+            let mut columns_slice = result.physical_columns.as_slice();
+            if self.col_len != columns_slice.len() {
+                for logical_row in &result.logical_rows {
+                    let mut data = vec![];
+                    columns_slice[0].encode(
+                        *logical_row,
+                        &field_type_from_column_info(&self.cols_info[0]),
```

Comment:
It is not a good idea to use this function frequently since it is not effective. I think you can change `encode` to accept `impl FieldTypeAccessor`?

Reply:
Done.